edb/server/args.py

```diff
@@ -402,23 +416,42 @@ def _writer(status: str) -> None:
              'no data dir. This option conflicts with --tls-cert-file and '
              '--tls-key-file, and defaults to True in dev mode.'),
     click.option(
-        '--allow-insecure-binary-clients',
-        envvar='EDGEDB_SERVER_ALLOW_INSECURE_BINARY_CLIENTS',
-        type=bool, is_flag=True, hidden=True,
-        help='Allow non-TLS client binary connections.'),
+        '--binary-endpoint-security',
+        envvar="EDGEDB_SERVER_BINARY_ENDPOINT_SECURITY",
+        type=click.Choice(
+            ['default', 'tls', 'optional'],
+            case_sensitive=True,
+        ),
+        default='default',
+        help='Specifies the security mode of server binary endpoint. '
+             'When set to `optional`, non-TLS connections are allowed. '
+             'The default is `tls`.',
+    ),
     click.option(
-        '--allow-insecure-http-clients',
-        envvar="EDGEDB_SERVER_ALLOW_INSECURE_HTTP_CLIENTS",
-        type=bool, is_flag=True, hidden=True,
-        help='Allow non-TLS client HTTP connections.'),
+        '--http-endpoint-security',
+        envvar="EDGEDB_SERVER_HTTP_ENDPOINT_SECURITY",
+        type=click.Choice(
+            ['default', 'tls', 'optional'],
+            case_sensitive=True,
+        ),
+        default='default',
+        help='Specifies the security mode of server HTTP endpoint. '
+             'When set to `optional`, non-TLS connections are allowed. '
+             'The default is `tls`.',
+    ),
     click.option(
-        '--insecure-dev-mode',
-        envvar="EDGEDB_SERVER_INSECURE_DEV_MODE",
-        type=bool, is_flag=True, hidden=True,
+        '--security',
+        envvar="EDGEDB_SERVER_SECURITY",
+        type=click.Choice(
+            ['default', 'strict', 'insecure_dev_mode'],
```

Comment:
```suggestion
            ['default', 'strict', 'insecure-dev-mode'],
```

to play nicely with the flag name -- `--insecure-dev-mode`?

Reply:
We decided on underscores in enum values in #3049